### PR TITLE
feat: show live $/% equivalent for tax, tip, and fees

### PR DESCRIPTION
**Pull request by jglopez**
## Summary

- Add `getAmountEquivalent` to `calculate.ts`: converts a tax/tip/fee amount string to its equivalent in the other unit ($ ↔ %) against the relevant dollar base, so a flat-dollar tip can be sanity-checked against the percentage it actually works out to.
- Show that equivalent next to each amount field in `TaxTipSection` (tax, tip, each additional fee), using a newly-exported `getTotalFeeBase` to resolve the right pre/post-tax base per field — same math `calculateBreakdown` already uses internally, no duplicated logic.
- Round percent equivalents to ~3 significant figures (decimal count floats with magnitude: 0 decimals ≥100%, 1 ≥10%, 2 ≥1%, 3 below that, capped at 3) instead of a flat 2 decimals, so small percentages don't lose all precision. Dollar equivalents stay at a flat 2 decimals.
- Only prefix with `~` when rounding actually lost precision — an exact match (e.g. `20%` of a base) shows plain, not `~20.00%`.

Display-only change: no edits to `BillState`, persistence, or settlement math.

## Test plan

- [x] New unit tests for `getAmountEquivalent` covering percent→dollar, dollar→percent, each sig-fig magnitude bucket, and zero/negative base edge cases (`npm test`, 16/16 passing)
- [x] TypeScript and build clean (`tsc --noEmit`, `npm run build`)
- [x] Verified base resolution against `calculateBreakdown` directly: a post-tax fee correctly uses subtotal+tax independent of the tip's own pre/post-tax setting; a zero-taxable-subtotal bill hides the $→% conversion but still shows %→$ as `$0.00`
- [x] Manual click-through in a real browser — not done, no browser tool available in this environment

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/calculate.ts [hunk 1]
```diff
@@ -39,6 +39,67 @@ export function isValidPrice(value: string): boolean {
...
+/**
+ * Round a dollar amount to 2 decimals, prefixed with "~" only when that
+ * rounding actually lost precision. Exact 2-decimal values are shown plain.
+ */
+function formatDollar(n: number): string {
+  const rounded = Math.round(n * 100) / 100
+  const isExact = Math.abs(n - rounded) < 1e-9
+  const prefix = isExact ? '' : '~'
+  return `${prefix}$${rounded.toFixed(2)}`
+}
```
[the remark this answers is not in the document]
Comment (jglopez) [reply src/utils/calculate.ts:69]: Good catch. Fixed in a7da10e06f846ed39a2a955448041c8861e743a8: `formatDollar` now separates the sign from the $, so negative amounts render as `-$5.00` instead of `$-5.00`.

File: src/utils/calculate.ts [hunk 2]
```diff
@@ -39,6 +39,67 @@ export function isValidPrice(value: string): boolean {
...
+  const isPercent = trimmed.endsWith('%')
+  if (isPercent) {
+    const pct = Number(trimmed.slice(0, -1))
+    if (isNaN(pct)) return null
+    return formatDollar((pct / 100) * base)
+  }
+  const n = Number(trimmed)
+  if (isNaN(n) || base <= 0) return null
+  return formatPercent((n / base) * 100)
+}
```
[the remark this answers is not in the document]
Comment (jglopez) [reply src/utils/calculate.ts:107]: Fixed in a7da10e06f846ed39a2a955448041c8861e743a8. A bare "%" now returns `null` (no digits to convert), and both the percent/dollar input and the base are checked with `isFinite` before doing any math, so `Infinity`/`NaN` inputs hide the equivalent instead of rendering `$Infinity` or `NaN%`.

File: src/utils/calculate.test.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  assertEqual(getAmountEquivalent('', 100), null, 'empty input has no equivalent')
+})
+
+test('invalid percent input -> hidden', () => {
+  assertEqual(getAmountEquivalent('abc%', 100), null, 'unparseable percent is hidden')
+})
+
+test('invalid dollar input -> hidden', () => {
+  assertEqual(getAmountEquivalent('abc', 100), null, 'unparseable dollar amount is hidden')
+})
```
[the remark this answers is not in the document]
Comment (jglopez) [reply src/utils/calculate.test.ts:117]: Added in a7da10e06f846ed39a2a955448041c8861e743a8: tests for bare "%" returning `null`, negative percent-to-dollar showing `-$5.00`, and non-finite percent/dollar/base inputs being hidden rather than producing `NaN`/`Infinity` strings.
